data/contract_docs/heatmap.md

```diff
@@ -0,0 +1,232 @@
+The true numeric range of each bucket can be indicated using an "le" label.  When absent,
+The field display is used for the bucket label.
+
+Example:
+
+<table>
+  <tr>
+    <td>
+      <strong>Type: Time</strong>
+      <p>
+        <strong>Name: Time</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: </strong>
+      </p>
+      <p>
+        <strong>Labels: {"le":<em> "10"</em>}</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: </strong>
+      </p>
+      <p>
+        <strong>Labels: {"le":<em> "20"</em>}</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: </strong>
+      </p>
+      <p>
+        <strong>Labels: {"le":<em> "+Inf"</em>}</strong>
+      </p>
+    </td>
+  </tr>
+  <tr>
+    <td>1653416391000</td>
+    <td>6</td>
+    <td>7</td>
+    <td>8</td>
+  </tr>
+  <tr>
+    <td>1653416391000</td>
+    <td>6</td>
+    <td>7</td>
+    <td>8</td>
+  </tr>
+  <tr>
+    <td>1653416391000</td>
+    <td>6</td>
+    <td>7</td>
+    <td>8</td>
+  </tr>
+</table>
+
+
+Note: [Timeseries wide](./timeseries.md#time-series-wide-format-timeserieswide) can be used directly
+as heatmap-buckets, in this case each value field becomes a row in the heatmap.
+
+
+## Heatmap scanlines (HeatmapScanlines)
+
+In this format, each row in the frame indicates the value of a single cell in a heatmap.
+There exists a row for every cell in the heatmap.
+
+**Example:**
+
+<table>
+  <tr>
+    <td>
+      <strong>Type: Time</strong>
+      <p>
+        <strong>Name: xMax|xMin|x</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: yMax|yMin|y</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: Count</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: Total</strong>
+      </p>
+    </td>
+    <td>
+      <strong>Type: Number</strong>
+      <p>
+        <strong>Name: Speed</strong>
+      </p>
+    </td>
+  </tr>
+  <tr>
+    <td>1653416391000</td>
+    <td>100</td>
+    <td>1</td>
+    <td>1</td>
+    <td>1</td>
+  </tr>
+  <tr>
+    <td>1653416391000</td>
+    <td>200</td>
+    <td>2</td>
+    <td>2</td>
+    <td>2</td>
+  </tr>
+  <tr>
+    <td>1653416391000</td>
+    <td>300</td>
+    <td>3</td>
+    <td>3</td>
+    <td>3</td>
+  </tr>
+
+  <tr>
+    <td>1653416392000</td>
+    <td>100</td>
+    <td>4</td>
+    <td>4</td>
+    <td>4</td>
+  </tr>
+  <tr>
+    <td>1653416392000</td>
+    <td>200</td>
+    <td>5</td>
+    <td>5</td>
+    <td>5</td>
+  </tr>
+  <tr>
+    <td>1653416392000</td>
+    <td>300</td>
+    <td>6</td>
+    <td>6</td>
+    <td>6</td>
+  </tr>
+</table>
+
+This format requires uniform cell sizing
```

Comment:
I don't understand want this means...? Perhaps if expressed in terms of fields/rows?

Reply:
@ryantxu Perhaps, "This format requires uniform cell sizing. The size of the cell is defined by the columns in each row that are chosen as the xMax|xMin|x and the yMax|yMin|y. We can see that the Number column(yMax|yMin|y) increases by 100(each cell is roughly 100 higher than the previous cell on the y axis) for each row containing a similar Time value(these stacked cells all have roughly the same location along the x axis). This produces a uniform cell size."

Reply:
@bohandley -- i just got everything in sync.  Please work with @leeoniya to figure out what is accurate and feel free to push to this branch and update whatever we need to make it more clear